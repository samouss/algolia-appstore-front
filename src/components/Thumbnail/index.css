.Thumbnail {
  width: 37px;
  height: 37px;
  background-color: #f9f9f9;
  overflow: hidden;
  border-radius: 17.544%;

  & img {
    width: 100%;
    height: auto;
  }

  @media (--tablet) {
    width: 57px;
    height: 57px;
  }
}
