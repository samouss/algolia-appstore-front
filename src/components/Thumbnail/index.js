import React from 'react';
import PropTypes from 'prop-types';
import './index.css';

const Thumbnail = ({ source, alt, ...props }) => (
  <div styleName="Thumbnail">
    <img
      {...props}
      src={source}
      alt={alt}
    />
  </div>
);

Thumbnail.propTypes = {
  source: PropTypes.string.isRequired,
  alt: PropTypes.string.isRequired,
};

export default Thumbnail;
