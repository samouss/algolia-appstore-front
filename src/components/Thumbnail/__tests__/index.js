import React from 'react';
import { shallow } from 'enzyme';
import Thumbnail from '../index';

describe('<Thumbnail />', () => {
  const defaultProps = {
    source: 'http://link.com',
    alt: 'Image alt',
    title: 'Image title',
  };

  it('expect to render', () => {
    const props = {
      ...defaultProps,
    };

    const component = shallow(
      <Thumbnail
        {...props}
      />,
    );

    expect(component).toMatchSnapshot();
  });
});
