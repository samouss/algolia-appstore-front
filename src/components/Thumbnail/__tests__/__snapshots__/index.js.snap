// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Thumbnail /> expect to render 1`] = `
<div
  styleName="Thumbnail"
>
  <img
    alt="Image alt"
    src="http://link.com"
    title="Image title"
  />
</div>
`;
